sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/XmlParserHelper.java

```diff
@@ -60,17 +66,43 @@ void checkRootTag(String name) {
   String nextStartTag() {
     try {
       while (stream.hasNext()) {
-        if (stream.next() == XMLStreamConstants.START_ELEMENT) {
+        if (getNextValid() == XMLStreamConstants.START_ELEMENT) {
           return stream.getLocalName();
         }
       }
-
       return null;
     } catch (XMLStreamException e) {
       throw new IllegalStateException("Error while parsing the XML file: " + file.getAbsolutePath(), e);
     }
   }
 
+  private int getNextValid() throws XMLStreamException {
+    Location lastLocation = stream.getLocation();
+    while (stream.hasNext()) {
+      try {
+        return stream.next();
+      } catch (XMLStreamException e) {
+        Location currentLocation = stream.getLocation();
+        if (isSameLocation(lastLocation, currentLocation)) {
+          // if the next() method throws exception before moving XML pointer forward, we fail here
+          LOG.warn("Unable to get next XML event while parsing file '{}'", file.toString());
+          throw e;
+        }
+        lastLocation = currentLocation;
+      }
+    }
+    return -1;
```

Comment:
this line could be covered by adding a test with an empty file

Reply:
Empty file actually does not work here for testing this part. It actually throws an exception when encountering an unexpected EOF token and does not put the reader pointer forward, so the exception gets rethrown as it falls into the case handled by the catch clause.
I added mock tests instead for coverage